functions/Get-DbaTable.ps1

```diff
@@ -77,126 +77,146 @@ Returns information on the CommandLog table in the DBA database on both instance
+					$splitDb = $splitDb.Substring(1, ($splitDb.Length - 2))
+				}
+
+				if ($schema -like "[[]*[]]") {
+					$schema = $schema.Substring(1, ($schema.Length - 2))
+				}
+
+				if ($tbl -like "[[]*[]]") {
+					$tbl = $tbl.Substring(1, ($tbl.Length - 2))
+				}
 
 				$fqtn = [PSCustomObject] @{
-					Database = $database
+					Database = $splitDb
 					Schema   = $Schema
 					Table    = $tbl
 				}
 				$fqtns += $fqtn
 			}
 		}
 	}
-	
+
 	process {
-		foreach ($instance in $sqlinstance) {	
+		foreach ($instance in $sqlinstance) {
 			try {
 				Write-Message -Level Verbose -Message "Connecting to $instance"
 				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
 			}
 			catch {
 				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
 			}
-			
-			#If IncludeSystemDBs is true, include systemdbs
-			#only look at online databases (Status equal normal)
+
 			try {
-				if ($Database) {
-					$dbs = $server.Databases | Where-Object { $Database -contains $_.Name -and $_.status -eq 'Normal' }
-				}
-				elseif ($IncludeSystemDBs) {
-					$dbs = $server.Databases | Where-Object { $_.status -eq 'Normal' }
+				#only look at online databases (Status equal normal)
+				$dbs = $server.Databases | Where-Object { $_.Status -eq 'Normal' }
+
+				#If IncludeSystemDBs is false, exclude systemdbs
+				if (!$IncludeSystemDBs) {
+					$dbs = $dbs | Where-Object { !$_.IsSystemObject }
 				}
-				else {
-					$dbs = $server.Databases | Where-Object { $_.status -eq 'Normal' -and $_.IsSystemObject -eq 0 }
+
+				if ($Database) {
+					$dbs = $dbs | Where-Object { $Database -contains $_.Name }
 				}
-				
+
 				if ($ExcludeDatabase) {
 					$dbs = $dbs | Where-Object { $ExcludeDatabase -notcontains $_.Name }
 				}
 			}
 			catch {
 				Stop-Function -Message "Unable to gather dbs for $instance" -Target $instance -Continue -InnerErrorRecord $_
 			}
-			
+
 			foreach ($db in $dbs) {
 				Write-Message -Level Verbose -Message "Processing $db"
-				
-				$d = $server.Databases[$db]
-				
+				$skipped = $false
+
 				if ($fqtns.Count -gt 0) {
 					foreach ($fqtn in $fqtns) {
-						if ($fqtn.schema -ne $NULL) {
-							try {
-								$tables = $db.Tables | Where-Object { $_.name -eq $tbl -and $_.Schema -eq $schema }
+						# If the user specified a database in a three-part name, and it's not the
+						# database currently being processed, skip this table.
+						if ($fqtn.Database) {
+							if ($fqtn.Database -ne $db.Name) {
+								$skipped = $true
+								continue
 							}
-							catch {
-								Write-Message -Level Warning -Message "Could not find table name: $($fqtn.tbl) schema: $($fqtn.schema)" -ErrorRecord $_
+							else {
+								$skipped = $false
 							}
 						}
-						else {
-							try {
-								$tables = $db.Tables | Where-Object { $_.name -eq $tbl }
+
+						$tables = $db.Tables | Where-Object { $_.Name -eq $fqtn.Table }
```

Comment:
This is a valid solution, @galador , but it's probably faster to avoid a few `if` and pipe to `Where` only once.

Something like this, maybe:
`$tables = $db.tables | Where-Object { $_.name -eq $fqtn.tbl -and $fqtn.schema -in ($schema,$null) -and $fqtn.database -in ($_.Parent.Name,$null)}`